src/components/FriendList/FriendSearch.js

```diff
@@ -0,0 +1,131 @@
+import * as React from "react";
+import SearchIcon from '@mui/icons-material/Search';
+import {InputAdornment} from "@mui/material";
+import TextField from '@mui/material/TextField';
+import Typography from "@mui/material/Typography";
+import Button from "@mui/material/Button";
+import AddIcon from "@mui/icons-material/Add";
+import {StudyRoomCard} from "../StudyRoom/CommonResources";
+import CheckIcon from '@mui/icons-material/Check';
+import axios from "axios";
+import {useNavigate} from "react-router";
+
+const ownerEmail = JSON.parse(localStorage.getItem("email"));
+const ownerUsername = JSON.parse(localStorage.getItem("username"));
+
+export default function FriendSearch() {
+    const [added, setAdded] = React.useState(false);
+    const [found, setFound] = React.useState(false);
+    const [searchName, setSearchName] = React.useState(null);
+    const [friend, setFriend] = React.useState('');
+    const [list, setList] = React.useState('');
+    const [text, setText] = React.useState('no one found');
+    const navigate = useNavigate();
+
+    React.useEffect(() => {
+        //user needs to be logged in to access
+        if (ownerEmail === undefined || ownerEmail === '') {
+            navigate("/login");
+        }
+        axios.get(`${process.env.REACT_APP_BASE_URL}friend/${ownerEmail}`)
+            .then(res => {
+                setList(res.data);
+            })
+            .catch(err => {
+                console.log('Error', err);
+            })
+
+    }, [text])
+
+    //send a friend request
+    function handleAdd(e) {
+        e.preventDefault();
+        setAdded(true);
+        axios.post(`${process.env.REACT_APP_BASE_URL}friend/add`, {senderEmail: ownerEmail, receiverEmail: friend})
+            .then(() => {
+                console.log('sent');
+            })
+            .catch(err => {
+                console.log('Error:', err)
+            });
+        window.location.reload();
```

Comment:
instead of forcing the reload, you can use a use state variable to force the re render of the component. Its way more effective.

Reply:
you can put this function as a dependency of the useEffect